yarn-project/validator-client/src/validator.ts

```diff
@@ -108,6 +146,46 @@ export class ValidatorClient extends WithTracer implements Validator {
     return this.validationService.attestToProposal(proposal);
   }
 
+  /**
+   * Re-execute the transactions in the proposal and check that the state updates match the header state
+   * @param proposal - The proposal to re-execute
+   */
+  async reExecuteTransactions(proposal: BlockProposal) {
+    const { header, txHashes } = proposal.payload;
+
+    const txs = (await Promise.all(txHashes.map(tx => this.p2pClient.getTxByHash(tx)))).filter(
+      tx => tx !== undefined,
+    ) as Tx[];
+
+    // If we cannot request all of the transactions, then we should fail
+    if (txs.length !== txHashes.length) {
+      this.log.error(`Failed to get transactions from the network: ${txHashes.join(', ')}`);
+      throw new TransactionsNotAvailableError(txHashes);
+    }
+
+    // Assertion: This check will fail if re-execution is not enabled
+    if (this.blockBuilder === undefined) {
+      throw new BlockBuilderNotProvidedError();
+    }
+
+    // We sync the state to the previous block as the public processor will not process the current block
+    const targetBlockNumber = header.globalVariables.blockNumber.toNumber() - 1;
+    this.log.verbose(`Re-ex: Syncing state to block number ${targetBlockNumber}`);
```

Comment:
super nit, but feels like this log belongs over in the sequencer, since it is the one actually doing the syncing.